internal/storage/types/global.go

```diff
@@ -0,0 +1,57 @@
+package types
+
+import (
+	"github.com/kubesimplify/ksctl/pkg/helpers/consts"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+type CredentialsDocument struct {
+	ID    primitive.ObjectID `bson:"_id,omitempty"`
+	Azure *CredentialsAzure  `bson:"azure,omitempty"`
+	Civo  *CredentialsCivo   `bson:"civo,omitempty"`
+}
```

Comment:
I have created a credential store document
question is how sould we use it
should we store a single document and everytime a fetch for credentials it gets the credentials and in the provider side it decides which credential it wants

or should we make credentials/civo.json , credentials/azure.json, etc. and it has info for that specific provider

what do you say?

Reply:
It has a reference https://github.com/kubesimplify/ksctl/pull/255#discussion_r1438551550